src/pages/index.js

```diff
@@ -126,18 +175,63 @@ const closeModal = () => {
 
 const handleEditFormSubmit = (event) => {
   event.preventDefault();
-  profileName.textContent = nameInput.value;
-  profileDescription.textContent = descriptionInput.value;
-  closeModal();
+  const submitBtn = event.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal();
+    })
+    .catch((err) => {
+      console.error(err).finally(() => {
+        setButtonText(submitBtn, false);
+      });
```

Comment:
* When I edit the profile data, the buttno text is always `Saving...` https://snipboard.io/MYtmdB.jpg 

`.finally` should be in line 192 (not after `console.error`)


```js
    .catch((err) => {
      console.error(err);
    })
    .finally(() => {
      setButtonText(submitBtn, false);
    });

```